bindings/python/dlite-python.i

```diff
@@ -931,33 +931,37 @@ int dlite_swig_set_scalar(void *ptr, DLiteType type, size_t size, obj_t *obj)
       void *p;
       if (SWIG_IsOK(SWIG_ConvertPtr(obj, &p, SWIGTYPE_p__Triple, 0))) {
         DLiteRelation *src = (DLiteRelation *)p;
-        triple_reset(ptr, src->s, src->p, src->o, src->id);
+        triple_reset(ptr, src->s, src->p, src->o, src->d, src->id);
 
       } else if (PySequence_Check(obj) || PyIter_Check(obj)) {
         int i;
         Py_ssize_t n;
         PyObject *lst;
-        char **s = ptr;  /* cast DLiteRelation to 4 string pointers */
-        assert(sizeof(DLiteRelation) == 4*sizeof(char *));
+        char **s = ptr;  /* cast DLiteRelation to 5 string pointers */
+        assert(sizeof(DLiteRelation) == 5*sizeof(char *));
 
-        /* Check that `obj` is a sequence of 3 or 4 strings */
+        /* Check that `obj` is a sequence of 3 to 5 strings */
         if (!(lst = PySequence_Fast(obj, "not a sequence or iterable")))
           FAIL("expected relation to be represented by a sequence or iterable");
 
-        if ((n = PySequence_Fast_GET_SIZE(lst)) < 3 || n > 4) {
+        if ((n = PySequence_Fast_GET_SIZE(lst)) < 3 || n > 5) {
           Py_DECREF(lst);
-          FAIL1("relations must be 3 or 4 strings, got %ld", (long)n);
+          FAIL1("relations must be 3 to 5 strings, got %ld", (long)n);
         }
         for (i=0; i<n; i++) {
           PyObject *item = PySequence_Fast_GET_ITEM(lst, i);
           if (!PyUnicode_Check(item)) {
             Py_DECREF(lst);
-            FAIL("relation subject, predicate and object must be strings");
+            FAIL("relation (s,p,o[,d[,id]]) items must be strings");
```

Comment:
I liked that the were written out in the previous error message. How about using subject(s), predicate(p)...